### PR TITLE
[BEAM-269] Create BigDecimal Coder

### PR DESCRIPTION
**Pull request by eljefe6a**
Jira issue BEAM-269 Create BigDecimal Coder. Added coder for BigDecimal. Added unit tests for BigDecimal coder.

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigDecimalCoder.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      int bigIntBytesSize = dataInputStream.readInt();
+
+      byte[] bigIntBytes = new byte[bigIntBytesSize];
+      dataInputStream.readFully(bigIntBytes);
+
+      BigInteger bigInteger = new BigInteger(bigIntBytes);
+      BigDecimal bigDecimal = new BigDecimal(bigInteger, scale);
+
+      return bigDecimal;
+    } catch (EOFException | UTFDataFormatException exn) {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigDecimalCoder.java]: I don't think you want to convert these, actually. They may indicate issues in the underlying stream.

Comment (eljefe6a) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigDecimalCoder.java]: I don't understand what you mean. What shouldn't be converted and what could be the potential issues?

Comment (eljefe6a) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigDecimalCoder.java]: I think I see what you're saying now. I shouldn't be doing a try/catch there. Will remove it.

I followed `BigEndianIntegerCoder` as an example. It does a try/catch for the same exceptions. A quick look found some other primitive coders doing the same thing. I don't know if those are an issue too.

**Pull request closed without merging**